src/theory/sets/theory_sets_rels.cpp

```diff
@@ -863,7 +864,7 @@ void TheorySetsRels::check(Theory::Effort level)
     sendInfer(fact, reason, "JOIN-Split-1");
     fact = NodeManager::currentNM()->mkNode(kind::MEMBER, mem2, join_rel[1]);
     sendInfer(fact, reason, "JOIN-Split-2");
-    makeSharedTerm( shared_x );
+    makeSharedTerm(shared_x, shared_type);
```

Comment:
This is safe because `shared_type` (which is the type of `shared_x`) is extracted from a set type in line [824](https://github.com/mudathirmahgoub/CVC4/blob/removeSubtyping/src/theory/sets/theory_sets_rels.cpp#L824).